### PR TITLE
Bug 31463811: Servicing: Redist (Nuget) conflicts with Microsoft.AI.MachineLearning starting 21H1+

### PR DESCRIPTION
**Pull request by smk2007**
Bug 31463811: Servicing: Redist (Nuget) conflicts with Microsoft.AI.MachineLearning starting 21H1+

WinML uses LoadLibrary to load onnxruntime.dll and directml.dll.

However, the loading of the library should be sensitive to where the calling library is located.
Currently:
  LoadLibraryExA("onnxruntime.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);

However, on Desktop this call will correctly load from System32 but will suffer from ordering issues if the redist is loaded first. If the redist is loaded first, the LoadLibraryExA call will resolve “onnxruntime.dll” to the one loaded in memory from outside of system32, and ignore the flags LOAD_LIBRARY_SEARCH_SYSTEM32.

In Store Apps this call will completely ignore the LOAD_LIBRARY_SEARCH_SYSTEM32 flag, and only load from the app package (if the redist binary - any binary with the same name - is present there). This causes the binary to be incorrectly loaded from System32 and load onnxruntime from the app package.

### CODE REVIEW

File: winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp [hunk 1]
```diff
@@ -16,30 +16,38 @@ static bool debug_output_ = false;
...
-static bool IsCurrentModuleInSystem32() {
-  std::string current_module_path;
-  current_module_path.reserve(MAX_PATH);
-  auto size_module_path = GetModuleFileNameA((HINSTANCE)&__ImageBase, current_module_path.data(), MAX_PATH);
-  FAIL_FAST_IF(size_module_path == 0);
-
+static std::string System32Path() {
   std::string system32_path;
   system32_path.reserve(MAX_PATH);
   auto size_system32_path = GetSystemDirectoryA(system32_path.data(), MAX_PATH);
```
Comment (ryanlai2) [inline winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp:22]: GetSystemDirectoryA handles the scenario where SysWOW64 folder is needed right?

**Review by ryanlai2 — DISMISSED** [review]

File: winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp (hunk 1 above)
Comment (smk2007) [reply winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp:22]: I think GetSystemWow64Directory does that... testing this now.

Comment (smk2007) [reply winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp]: The first iteration doesnt work for WOW64.. fixed.

Now trying to resolve WinSxS.

Comment (smk2007) [reply winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp]: ACTUALLY, so LoadLibraryExA("C:\\windows\\system32\\...") will automagically resolve to syswow64 when needed!

Comment (ryanlai2) [reply winml/lib/Api.Ort/OnnxruntimeEnvironment.cpp]: Nicee!!

**Review by ryanlai2 — DISMISSED** [review]

**Review by ryanlai2 — APPROVED** [review]

**Comment by snnn** [comment]
@liqunfu , could you please help take a look? This change is Windows only but the Nuphar build has a segfault.

**Pull request merged**